doc/https-certificates.md

```diff
@@ -0,0 +1,442 @@
+
+    // Create certificate object
+    X509Certificate2 certificate = new X509Certificate2(
+        Encoding.UTF8.GetBytes(serverCrt), 
+        Encoding.UTF8.GetBytes(serverKey), 
+        "1234"  // Password used to encrypt the private key
+    );
+
+    // Create HTTPS server
+    using (WebServer server = new WebServer(443, HttpProtocol.Https, new Type[] { typeof(MyController) }))
+    {
+        server.HttpsCert = certificate;
+        server.SslProtocols = SslProtocols.Tls12;
+        
+        server.Start();
+        Thread.Sleep(Timeout.Infinite);
+    }
+}
+```
+
+## Complete HTTPS Example
+
+```csharp
+public class SecureController
+{
+    [Route("secure/data")]
+    [Authentication("Basic")]
+    public void GetSecureData(WebServerEventArgs e)
+    {
+        var data = new 
+        { 
+            message = "Secure data over HTTPS", 
+            timestamp = DateTime.UtcNow,
+            encrypted = true
+        };
+        
+        string json = JsonConvert.SerializeObject(data);
+        e.Context.Response.ContentType = "application/json";
+        WebServer.OutPutStream(e.Context.Response, json);
+    }
+}
+
+public static void Main()
+{
+    // Connect to WiFi
+    var connected = WifiNetworkHelper.ConnectDhcp(ssid, password, requiresDateTime: true);
+    if (!connected) return;
+
+    // Load certificate
+    X509Certificate2 cert = LoadCertificate();
+
+    using (WebServer server = new WebServer(443, HttpProtocol.Https, new Type[] { typeof(SecureController) }))
+    {
+        // Configure HTTPS
+        server.HttpsCert = cert;
+        server.SslProtocols = SslProtocols.Tls12;
+        
+        // Configure authentication
+        server.Credential = new NetworkCredential("admin", "securepass");
+        
+        server.Start();
+        Debug.WriteLine("HTTPS server started on port 443");
+        Thread.Sleep(Timeout.Infinite);
+    }
+}
+
+private static X509Certificate2 LoadCertificate()
+{
+    // Certificate content (replace with your actual certificate)
+    const string certContent = @"-----BEGIN CERTIFICATE-----
+... your certificate content here ...
+-----END CERTIFICATE-----";
+
+    const string keyContent = @"-----BEGIN RSA PRIVATE KEY-----
+... your encrypted private key here ...
+-----END RSA PRIVATE KEY-----";
+
+    return new X509Certificate2(
+        Encoding.UTF8.GetBytes(certContent),
+        Encoding.UTF8.GetBytes(keyContent),
+        "your-private-key-password"
+    );
+}
+```
+
+## SSL/TLS Protocol Configuration
+
+This section goes through different options for the SSL configuration.
+
+### TLS 1.2 Only (Recommended)
+
+```csharp
+server.SslProtocols = SslProtocols.Tls12;
+```
+
+### Multiple TLS Versions
+
+```csharp
+server.SslProtocols = SslProtocols.Tls | SslProtocols.Tls11 | SslProtocols.Tls12;
+```
+
+### Available Options
+
+- `SslProtocols.Tls` - TLS 1.0
+- `SslProtocols.Tls11` - TLS 1.1  
+- `SslProtocols.Tls12` - TLS 1.2 (recommended)
+
+## Certificate Storage Options
```

Comment:
Missing here the upload from VS extension -> network options. Available for root CA and device cert.